### PR TITLE
Update progress indicator API docs

### LINKED ISSUE #13630: CircularProgressIndicator should provide a simple color setting
## Steps to Reproduce

```dart
new CircularProgressIndicator(
        valueColor: new AlwaysStoppedAnimation<Color>(Colors.green)
```

I had search quite a bit to find the information in gitter how to change the progress bar color and wondered why there is no direct setting for the color to make this easier.

## Flutter Doctor
```

C:\Projects\emoji>flutter doctor
[√] Flutter (on Microsoft Windows [Version 10.0.16299.125], locale en-US, channel alpha)
    • Flutter at c:\sdks\flutter
    • Framework revision 8f65fec5f5 (3 days ago), 2017-12-12 09:50:14 -0800
    • Engine revision edaecdc8b8
    • Tools Dart version 1.25.0-dev.11.0
    • Engine Dart version 2.0.0-edge.d8ae797298c3a6cf8dc9f4558707bd2672224d3e

[√] Android toolchain - develop for Android devices (Android SDK 27.0.1)
    • Android SDK at C:\Users\ride4\AppData\Local\Android\sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-27, build-tools 27.0.1
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-915-b01)

[√] Android Studio (version 3.0)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-915-b01)

[√] IntelliJ IDEA Community Edition (version 2017.2)
    • Flutter plugin version 19.1
    • Dart plugin version 172.4343.25

[√] Connected devices
    • Android SDK built for x86 • emulator-5554 • android-x86 • Android 7.1.1 (API 25) (emulator)
```

### PR DESCRIPTION
**Pull request by HansMuller**
Fixes https://github.com/flutter/flutter/issues/13630

Cleaned up the progress indicator doc a little and mentioned `AlwaysStoppedAnimation<Color>(...)` in the class docs for LinearProgressIndicator, CircularProgressIndicator.

### CODE REVIEW

**Comment by ride4sun** [comment]
I disagree - from the user perspective, it takes several minutes if not longer to find the docs for this obscure setting. As a developer, I start with the IDE (Intelly Sense) dropdown help. Then I have startup the docs... etc. Adding a shorthand would probably take less than a minute to add but would save many people valuable time down the road and adds consistency throughout the framework. Just my 2 Cents.

**Comment by HansMuller** [comment]
The doc for the `valueColor` property, one of only a handful, explains using AlwaysStoppedAnimationColor for a constant color. Likewise for the CircularProgressIndicator class doc.  It shouldn't take several minutes to sort that out (anymore).

File: packages/flutter/lib/src/material/progress_indicator.dart [hunk 1]
```diff
@@ -48,21 +48,23 @@ abstract class ProgressIndicator extends StatefulWidget {
...
-  /// The progress indicator's background color. The current theme's
-  /// [ThemeData.backgroundColor] by default.
+  /// The progress indicator's background color.
+  ///
+  /// The current theme's [ThemeData.backgroundColor] by default.
   final Color backgroundColor;
 
-  /// The indicator's color is the animation's value. To specify a constant
-  /// color use: `new AlwaysStoppedAnimation<Color>(color)`.
+  /// The indicator's color as an animated value.
```
Comment (gspencergoog) [inline packages/flutter/lib/src/material/progress_indicator.dart]: This should either be "progress indicator's", or the previous one should be "indicator's" for consistency.

**Review by gspencergoog — APPROVED** [review]
32384589-a60f0e74-c078-11e7-9bc1-e5b5287aea9d

File: packages/flutter/lib/src/material/progress_indicator.dart (hunk 1 above)
Comment (HansMuller) [reply packages/flutter/lib/src/material/progress_indicator.dart]: Sounds good, done.

**Pull request merged**